cmd/cloud_sql_proxy/cloud_sql_proxy.go

```diff
@@ -626,6 +636,10 @@ func main() {
 
 	logging.Infof("Ready for new connections")
 
+	if *useHealthCheck {
+		healthcheck.NotifyReadyForConnections(hc)
```

Comment:
Since we already have a reference to `hc`, this should be a method on the `HealthCheck` type:

``` golang
hc.NotifyReadyForConnections()
```

Then in the body of that method, you can check if `hc` is nil and avoid the duplicate `*useHealthCheck` check (Go lets you call methods on nil receivers).